transport/http/jsonrpc/request_response_types.go

```diff
@@ -0,0 +1,28 @@
+package jsonrpc
+
+import "encoding/json"
+
+// Request defines a JSON RPC request from the spec
+// http://www.jsonrpc.org/specification#request_object
+type Request struct {
+	JSONRPC string          `json:"jsonrpc"`
+	Method  string          `json:"method"`
+	Params  json.RawMessage `json:"params"`
+	ID      interface{}     `json:"id"`
```

Comment:
Instead of `interface{}`, the ID could be a custom type that implements MarshalJSON. Thoughts?